build/localnet/cluster-manager/sts_kill.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+// Monitors Pods created by StatefulSets, and if Pods are in `CrashLoopBackOff` status,
+// and they have different image tag - kill them so StatefulSet would recreate the Pod with new image.
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	pocketk8s "github.com/pokt-network/pocket/shared/k8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	watch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	appstypedv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coretypedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// Loop through existing pods and set up a watch for new pods so we don't hit Kubernetes API all the time
+func initDeleteCrashedPods(client *kubernetes.Clientset) {
+	stsClient := client.AppsV1().StatefulSets(pocketk8s.CurrentNamespace)
+	podClient := client.CoreV1().Pods(pocketk8s.CurrentNamespace)
+
+	// Loop through all existing Pods and delete the ones that are in CrashLoopBackOff status
+	podList, _ := podClient.List(context.TODO(), metav1.ListOptions{})
+	for i := range podList.Items {
```

Comment:
@Olshansk that's what I originally did, but [gocritic complained](https://go-critic.com/overview.html#rangevalcopy). IMO, this is a side script that does not require great performance, but I did want to pass the gate checks. :)

Reply:
Should we not be doing `pod := &podList.Items[i]` then?

Ditto elsewhere

Reply:
Don't know if this makes a difference, but let me change that to the example from gocritic, good call.

Reply:
Can you make this change here and elsewhere?

Here is the reason I ask:

```go
	for i := range podList.Items {
		pod := podList.Items[i]
		if err := deleteCrashedPods(&pod, stsClient, podClient); err != nil {
```

I chatGPTed why anyone would do this and the reasoning is to avoid copying values. You see that you're using a pointer on the third line so doing `pod := &podList.Items[i]` helps me (as the reader of the code) understand why you didn't do `for I, pod := range podList.Items {`.

Please update it here and elsewhere.

Reply:
@Olshansk thanks, that makes sense! I was thinking I had updated them, but I missed one line. I think they are all pointers now.